test/unit/Orchestrator.js

```diff
@@ -0,0 +1,358 @@
+const MockDownstream = artifacts.require('MockDownstream.sol');
+const MockUFragmentsPolicy = artifacts.require('MockUFragmentsPolicy.sol');
+const Orchestrator = artifacts.require('Orchestrator.sol');
+const RebaseCallerContract = artifacts.require('RebaseCallerContract.sol');
+const ConstructorRebaseCallerContract = artifacts.require('ConstructorRebaseCallerContract.sol');
+
+const BigNumber = web3.BigNumber;
+const _require = require('app-root-path').require;
+const BlockchainCaller = _require('/util/blockchain_caller');
+const chain = new BlockchainCaller(web3);
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+let orchestrator, mockPolicy, mockDownstream;
+let r;
+let deployer, user;
+
+async function setupContracts () {
+  await chain.waitForSomeTime(86400);
+  const accounts = await chain.getUserAccounts();
+  deployer = accounts[0];
+  user = accounts[1];
+  mockPolicy = await MockUFragmentsPolicy.new();
+  orchestrator = await Orchestrator.new(mockPolicy.address);
+  mockDownstream = await MockDownstream.new();
+}
+
+contract('Orchestrator', function (accounts) {
+  before('setup Orchestrator contract', setupContracts);
+
+  describe('when sent ether', async function () {
+    it('should reject', async function () {
+      expect(
+        await chain.isEthException(orchestrator.sendTransaction({ from: user, value: 1 }))
+      ).to.be.true;
+    });
+  });
+
+  describe('when rebase called by a contract', function () {
+    it('should fail', async function () {
+      const rebaseCallerContract = await RebaseCallerContract.new();
+      expect(
+        await chain.isEthException(rebaseCallerContract.callRebase(orchestrator.address))
+      ).to.be.true;
+    });
+  });
+
+  describe('when rebase called by a contract which is being constructed', function () {
+    it('should fail', async function () {
+      expect(
+        await chain.isEthException(ConstructorRebaseCallerContract.new(orchestrator.address))
+      ).to.be.true;
+    });
+  });
+
+  describe('when transaction list is empty', async function () {
+    before('calling rebase', async function () {
+      r = await orchestrator.rebase();
+    });
+
```

Comment:
assert the length of `r.receipt.logs` is 1

Reply:
Actually I agree :) Added checks.